feat: introduced LearningHeader for frontend-app-learning

asadiqbal08 opened the pull request:
fixes: https://github.com/mitodl/mitxonline/issues/252

#### Earlier Review References of @davidjoy 
[Reference-1](https://github.com/mitodl/frontend-component-header-mitol/pull/1)
[Reference-2](https://github.com/openedx/build-test-release-wg/issues/81#issuecomment-928050901)

As per some suggestion by David, We need to refactor the code in EdX Learning MFE to utilize the header from frontend-component-header instead of course-header

![screencapture-localhost-2000-course-course-v1-edX-DemoX-Demo-Course-block-v1-edX-DemoX-Demo-Course-type-sequential-block-edx-introduction-block-v1-edX-DemoX-Demo-Course-type-vertical-block-vertical-0270f6de40fc-2021-10-22-17_28_58](https://user-images.githubusercontent.com/7334669/138453707-c713613e-794e-460f-907e-339081dc80be.png)
.

asadiqbal08 commented:
@davidjoy anything else need to address here or good to go ?

davidjoy commented:
Ugh, it still seems to have some package-lock.json problems.  Try running `npm ci` locally to see the error - looks like package-lock.json doesn't like some font-awesome versions.  Does running `npm i` produce any changes to package-lock.json?  If not, we may need to just re-generate package-lock.json by deleting it and running `npm i`, but it'd be preferable not to if we can avoid it. 

Otherwise this looks good to go and as soon as we get that check to pass, we can merge.

davidjoy reviewed (changes requested):
It looks like there are issues with package-lock.json - see my prior comment above.

asadiqbal08 commented:
Yes, I forgot to push updated `package.json`, Now I am not getting any error locally. let see over the CI pipeline here.

asadiqbal08 commented:
@davidjoy I updated the PR to fix issues that causing build failure in the CI pipeline. I make sure by re running the Github action on local devstack and also fixed the corresponding test failure issues. 🤞🤞

asadiqbal08 commented on src/Header.jsx:
```diff
@@ -57,18 +63,18 @@ function Header({ intl }) {
 
+  // Users should only see Order History if have a ORDER_HISTORY_URL define in the environment.
+  if (config.ORDER_HISTORY_URL) {
+    userMenu.splice(-1, 0, orderHistoryItem);
```
@davidjoy FYI, I kept the OrderHistory conditional on basis of `ORDER_HISTORY_URL`

asadiqbal08 commented on src/learning-header/AuthenticatedUserDropdown.jsx:
```diff
@@ -0,0 +1,57 @@
+          <Dropdown.Item href={`${getConfig().LMS_BASE_URL}/account/settings`}>
+            {intl.formatMessage(messages.account)}
+          </Dropdown.Item>
+          { getConfig().ORDER_HISTORY_URL && (
```
Here too.

edx-webhook commented:
@asadiqbal08 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-component-header&entry.867055334=https://github.com/edx/frontend-component-header/pull/133&entry.1484655318=https://github.com/asadiqbal08&entry.752974735=2021-10-28+08:28&entry.1917517419=2021-12-02+19:17&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
